### PR TITLE
Support for Single line credit in Journal Entries

### PR DESCRIPTION
**Pull request by ashwin1111**

### CODE REVIEW

File: apps/quickbooks_online/utils.py [hunk 1]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
 
     @staticmethod
-    def __construct_journal_entry_lineitems(journal_entry_lineitems: List[JournalEntryLineitem],
-                                            posting_type) -> List[Dict]:
+    def __group_journal_entry_credits(journal_entry_lineitems: List[JournalEntryLineitem]) -> List[Dict]:
+        total_sum = sum(line.amount for line in journal_entry_lineitems)
+
+        return [{
+            'DetailType': 'JournalEntryLineDetail',
+            'Description': 'Total',
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:606]: Description TBD

File: apps/quickbooks_online/utils.py [hunk 2]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
+            'Amount': total_sum,
+            'JournalEntryLineDetail': {
+                'PostingType': 'Credit',
+                'AccountRef': {
+                    'value': journal_entry_lineitems[0].debit_account_id
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:611]: debit_account_id would be same for all lines

File: apps/quickbooks_online/utils.py [hunk 3]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
+                },
+                'DepartmentRef': {
+                    'value': journal_entry_lineitems[0].department_id
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:614]: TBD, if we can put 1st item's property or None

File: apps/quickbooks_online/utils.py [hunk 4]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
+                },
+                'ClassRef': {
+                    'value': journal_entry_lineitems[0].class_id
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:617]: TBD, if we can put 1st item's property or None

File: apps/quickbooks_online/utils.py [hunk 5]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
+                },
+                'Entity': {
+                    'EntityRef': {
+                        'value': journal_entry_lineitems[0].entity_id
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:621]: would be same, since we group by employee email by default

File: apps/quickbooks_online/utils.py [hunk 6]
```diff
@@ -598,17 +598,45 @@ def post_credit_card_purchase(self, credit_card_purchase: CreditCardPurchase,
...
+                    },
+                    'Type': journal_entry_lineitems[0].entity_type,
```
Comment (ashwin1111) [inline apps/quickbooks_online/utils.py:623]: should be same for all lines

File: apps/quickbooks_online/utils.py (hunk 1 above)
Comment (Shwetabhk) [reply apps/quickbooks_online/utils.py:606]: `Total Amount`

File: apps/quickbooks_online/utils.py (hunk 2 above)
Comment (Shwetabhk) [reply apps/quickbooks_online/utils.py:611]: ACK

File: apps/quickbooks_online/utils.py (hunk 3 above)
Comment (Shwetabhk) [reply apps/quickbooks_online/utils.py:614]: leave empty

File: apps/quickbooks_online/utils.py (hunk 4 above)
Comment (Shwetabhk) [reply apps/quickbooks_online/utils.py:617]: leave empty

**Review by Shwetabhk — APPROVED** [review]

**Pull request merged**